external_parser/event_processors/joined_event.h

```diff
@@ -300,6 +300,102 @@ struct ccb_joined_event : public typed_joined_event {
   }
 };
 
+struct DecisionServiceInteractionCats {
+  std::string eventId;
+  std::string timestamp;
+  float action;
+  float pdf_value;
+  float probabilityOfDrop = 0.f;
+  bool skipLearn{false};
+};
+
+struct ca_joined_event : public typed_joined_event {
+  DecisionServiceInteractionCats interaction_data;
+  float reward = 0.0f;
+  float original_reward = 0.0f;
+
+  ~ca_joined_event() = default;
+
+  bool is_skip_learn() const override { return interaction_data.skipLearn; }
+
+  void set_skip_learn(bool sl) override { interaction_data.skipLearn = sl; }
+
+  void set_apprentice_reward() override {
+    if (!std::isnan(interaction_data.action)) {
```

Comment:
q: how does apprentice mode work for ca? it doesn't need to match any baseline action?

Reply:
I'm not sure how to set up a baseline action here since the actions are continuous. @olgavrou any thoughts?

Reply:
baseline has not been implemented for CA so id say just do nothing here, `set_apprentice_reward` shouldn't be called in `calc_and_set_cost` `if (interaction_metadata.learning_mode == v2::LearningModeType_Apprentice)` should log that apprentice isn't implemented for CA and just do what you would have done if `learning_mode != apprentice`